### DISCUSSION TITLE
Network Time Security support

### DISCUSSION THREAD

**Discussion by EntropySmoke** [AdguardTeam/AdGuardHome#2813] [Ideas]
### Prerequisites
- [x] I am running the latest version
- [x] I checked the documentation and found no answer
- [x] I checked to make sure that this issue has not already been filed

### Problem Description
Network Time Protocol (NTP) amplification attacks are a reality and there is no reason not to use newer Network Time Security (NTS) from providers like Cloudflare. The problem is that most devices do not support NTS. AdGuard Home can distinguish itself by being one of the first tools to support NTS.

### Proposed Solution
Provide support for Network Time Security (NTS) and a list of providers that support NTS.

**Comment by bcookatpcsd** [comment]
(I am not an AdG developer..)

Are you looking for AdG Home to sync w/ an nts server or be an nts server?

https://fedoramagazine.org/secure-ntp-with-nts/

I saw this the other day and wanted to see about setting up my own nts server for internal use..

**Comment by EntropySmoke** [comment]
Synchronization with NTS is what I am looking for. I think it needs to happen over port 1234, at least for time.cloudflare.com.  For me, AdGuard Home does not utilize local time zone set in Raspberry Pi 2 configuration. Therefore, even if I could get NTS working with Raspberry Pi 2, it would have no effect on AdGuard Home time. That is why I ask to have NTS directly implemented in AdGuard Home. It can provide secure time synchronization.

**Comment by bcookatpcsd** [comment]
if you can get into your pi on the command line.. 

vi /etc/profile (and add I put it at the top, but that's me..)
TZ='America/New_York'; export TZ

vi /etc/environment
TZ='America/New_York'

or whatever your tz is.. 

Save, reboot.. see if that deals with sounds like log information.. 

https://blog.packagecloud.io/eng/2017/02/21/set-environment-variable-save-thousands-of-system-calls/

I've had tons of issues with $whatever b/c TZ wasn't set.. 

my 0.02; ymmv

`
984bce@arch0069-984bce /o/traefik> head /etc/profile
# /etc/profile

export TZ='America/New_York'

# Set our umask
umask 022

# Append our default paths
appendpath () {
    case ":$PATH:" in

984bce@arch0069-984bce /o/traefik> head /etc/environment 

TZ='America/New_York'

`

**Comment by ameshkov** [comment]
> For me, AdGuard Home does not utilize local time zone set in Raspberry Pi 2 configuration

Hm, well, but it should. We don't do anything specific with time zones, just use Golang's `time` struct which does use system time.

**Comment by EntropySmoke** [comment]
Raspberry Pi 2 always specifies 2 time zones - UTC and Local. Both are correct on my system, but AdGuard Home selects UTC time as the default one, regardless of the log setting.

**Comment by EntropySmoke** [comment]
This has never been resolved for me. My RP always had correct UTC (sudo date -u) and correct local time (sudo date), always synchronized, but AGH log always used UTC instead of local time.

**Reply by ameshkov** [reply]
Well, I suppose this is a Golang issue then.

I doubt we should add direct NTS sync functionality to AGH, this is what OS should do.
